### PR TITLE
feat(service): enable tracing on all services

### PR DESCRIPTION
**Pull request by fsamin**

### CODE REVIEW

File: engine/hatchery/swarm/swarm_util_create.go [hunk 1]
```diff
@@ -13,13 +13,16 @@ import (
...
+	"github.com/ovh/cds/engine/api/tracing"
 	"github.com/ovh/cds/sdk"
 	"github.com/ovh/cds/sdk/hatchery"
 	"github.com/ovh/cds/sdk/log"
 )
 
 //create the docker bridge
-func (h *HatcherySwarm) createNetwork(dockerClient *dockerClient, name string) error {
+func (h *HatcherySwarm) createNetwork(ctx context.Context, dockerClient *dockerClient, name string) error {
+	ctx, end := tracing.Span(ctx, "swarm.createNetwork", tracing.Tag("network", name))
```
Comment (golangci) [inline engine/hatchery/swarm/swarm_util_create.go]: ineffectual assignment to `ctx`

File: engine/hatchery/swarm/swarm_util_create.go [hunk 2]
```diff
@@ -46,7 +49,10 @@ type containerArgs struct {
 }
 
 //shortcut to create+start(=run) a container
-func (h *HatcherySwarm) createAndStartContainer(dockerClient *dockerClient, cArgs containerArgs, spawnArgs hatchery.SpawnArguments) error {
+func (h *HatcherySwarm) createAndStartContainer(ctx context.Context, dockerClient *dockerClient, cArgs containerArgs, spawnArgs hatchery.SpawnArguments) error {
+	ctx, end := tracing.Span(ctx, "swarm.createAndStartContainer", tracing.Tag(tracing.TagWorker, cArgs.name))
```
Comment (golangci) [inline engine/hatchery/swarm/swarm_util_create.go]: ineffectual assignment to `ctx`

File: sdk/hatchery/hatchery.go [hunk 3]
```diff
@@ -208,31 +214,64 @@ func Create(h Interface) error {
...
+					tracing.Tag(tracing.TagWorkflowNodeJobRun, j.ID),
+				)
+			}
+			endTrace := func(reason string) {
+				if reason != "" {
+					tracing.Current(currentCtx,
+						tracing.Tag("reason", reason),
+					)
+				}
+				tracing.End(currentCtx, nil, nil)
```
Comment (golangci) [inline sdk/hatchery/hatchery.go]: Error return value of `tracing.End` is not checked

File: sdk/hatchery/hatchery.go [hunk 4]
```diff
@@ -208,31 +214,64 @@ func Create(h Interface) error {
...
+					tracing.Tag(tracing.TagWorkflowNodeJobRun, j.ID),
+				)
+			}
+			endTrace := func(reason string) {
+				if reason != "" {
+					tracing.Current(currentCtx,
+						tracing.Tag("reason", reason),
+					)
+				}
+				_ = tracing.End(currentCtx, nil, nil)
```
Comment (golangci) [inline sdk/hatchery/hatchery.go]: cannot assign 2 values to 1 variables

File: engine/hatchery/swarm/swarm_util_create.go [hunk 5]
```diff
@@ -89,11 +95,13 @@ func (h *HatcherySwarm) createAndStartContainer(dockerClient *dockerClient, cArg
 		}
 	}
 
+	_ := next tracing.Span(ctx, "swarm.dockerClient.ImageList")
```
Comment (golangci) [inline engine/hatchery/swarm/swarm_util_create.go]: expected ';', found 'IDENT' tracing (and 3 more errors)

File: engine/hatchery/swarm/swarm_util_create.go [hunk 6]
```diff
@@ -89,11 +95,13 @@ func (h *HatcherySwarm) createAndStartContainer(dockerClient *dockerClient, cArg
...
 	// Check the images to know if we had to pull or not
-	images, errl := dockerClient.ImageList(context.Background(), types.ImageListOptions{All: true})
+	images, errl := dockerClient.ImageList(ctx, types.ImageListOptions{All: true})
 	if errl != nil {
 		log.Warning("createAndStartContainer> Unable to list images: %s", errl)
 	}
+	next()
```
Comment (golangci) [inline engine/hatchery/swarm/swarm_util_create.go]: undeclared name: next

File: engine/hatchery/swarm/swarm_util_create.go [hunk 7]
```diff
@@ -111,19 +119,28 @@ checkImage:
 	}
 
 	if !imageFound {
+		_ := next tracing.Span(ctx, "swarm.dockerClient.pullImage", tracing.Tag("image", cArgs.image))
```
Comment (golangci) [inline engine/hatchery/swarm/swarm_util_create.go]: undeclared name: next

File: engine/hatchery/swarm/swarm_util_create.go [hunk 8]
```diff
@@ -111,19 +119,28 @@ checkImage:
...
+		_, next := tracing.Span(ctx, "swarm.dockerClient.pullImage", tracing.Tag("image", cArgs.image))
 		if err := h.pullImage(dockerClient, cArgs.image, timeoutPullImage); err != nil {
+			next()
 			return sdk.WrapError(err, "createAndStartContainer> Unable to pull image %s on %s", cArgs.image, dockerClient.name)
 		}
+		next()
 	}
 
-	c, err := dockerClient.ContainerCreate(context.Background(), config, hostConfig, networkingConfig, name)
+	_, next:= tracing.Span(ctx, "swarm.dockerClient.ContainerCreate", tracing.Tag(tracing.TagWorker, cArgs.name), tracing.Tag("network", fmt.Sprintf("%v", networkingConfig))
```
Comment (golangci) [inline engine/hatchery/swarm/swarm_util_create.go]: missing ',' before newline in argument list (and 10 more errors)

File: engine/hatchery/swarm/swarm.go [hunk 9]
```diff
@@ -213,12 +221,12 @@ func (h *HatcherySwarm) SpawnWorker(spawnArgs hatchery.SpawnArguments) (string,
 				if network == "" {
 					network = name + "-net"
 					networkAlias = "worker"
-					if err := h.createNetwork(dockerClient, network); err != nil {
+					if err := h.createNetwork(ctx, dockerClient, network); err != nil {
```
Comment (golangci) [inline engine/hatchery/swarm/swarm.go]: invalid operation: h (variable of type *HatcherySwarm) has no field or method createNetwork

File: engine/hatchery/swarm/swarm.go [hunk 10]
```diff
@@ -268,7 +276,7 @@ func (h *HatcherySwarm) SpawnWorker(spawnArgs hatchery.SpawnArguments) (string,
 					entryPoint:   nil,
 				}
 
-				if err := h.createAndStartContainer(dockerClient, args, spawnArgs); err != nil {
+				if err := h.createAndStartContainer(ctx, dockerClient, args, spawnArgs); err != nil {
```
Comment (golangci) [inline engine/hatchery/swarm/swarm.go]: invalid operation: h (variable of type *HatcherySwarm) has no field or method createAndStartContainer

File: engine/hatchery/swarm/swarm.go [hunk 11]
```diff
@@ -386,7 +394,7 @@ func (h *HatcherySwarm) SpawnWorker(spawnArgs hatchery.SpawnArguments) (string,
 	}
 
 	//start the worker
-	if err := h.createAndStartContainer(dockerClient, args, spawnArgs); err != nil {
+	if err := h.createAndStartContainer(ctx, dockerClient, args, spawnArgs); err != nil {
```
Comment (golangci) [inline engine/hatchery/swarm/swarm.go]: invalid operation: h (variable of type *HatcherySwarm) has no field or method createAndStartContainer

**Comment by ovh-cds** [comment]
CDS Report ut-engine#5354.0 ✘  
* Stage 1
  * Engine Test with DB ✘

**Pull request merged**